File: sdks/ts/src/sdk/models/operations/getfilesidversionsid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetFilesIdVersionsIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_id" })
  fileId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=file_version_id" })
  fileVersionId: string;
}


export class GetFilesIdVersionsIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetFilesIdVersionsIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetFilesIdVersionsIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetFilesIdVersionsIdQueryParams;
}


export class GetFilesIdVersionsIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  fileVersion?: shared.FileVersion;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/getfilesidversionsid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetFilesIDVersionsIDPathParams struct {
	FileID        string `pathParam:"style=simple,explode=false,name=file_id"`
	FileVersionID string `pathParam:"style=simple,explode=false,name=file_version_id"`
}

type GetFilesIDVersionsIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetFilesIDVersionsIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetFilesIDVersionsIDPathParams
	QueryParams GetFilesIDVersionsIDQueryParams
}

type GetFilesIDVersionsIDResponse struct {
	ClientError *shared.ClientError
	ContentType string
	FileVersion *shared.FileVersion
	StatusCode  int64
}
